### PR TITLE
feat(FX-4041): Add active filter pills to partner page's artwork grid

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR solves [FX-4041]

Review app: https://selected-filters-for-partner.artsy.net

### Context
We’ve added active filter pills to the artist page’s artworks grid a while ago, which is also where the “Create alert” control lives. We’d like to first expand these active filter pills to other grids before we also support alert creation on those surfaces.

image

### Acceptance Criteria
* As a collector
* When I visit a partner’s artwork grid
* And I enable at least one filter
* Then I should see my active filters represented as pills above the artwork grid
* And "Create Alert" button should **NOT** be displayed

#### Quick links
* [Aicon Gallery](https://selected-filters-for-partner.artsy.net/partner/aicon-gallery/works)
* [Paragon](https://selected-filters-for-partner.artsy.net/partner/paragon/works)
* [Galerie Denise René](https://selected-filters-for-partner.artsy.net/partner/galerie-denise-rene/works)

### Demo

[FX-4041]: https://artsyproduct.atlassian.net/browse/FX-4041?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/v2/Components/ArtworkFilter/index.tsx [hunk 1]
```diff
@@ -275,10 +275,11 @@ export const BaseArtworkFilter: React.FC<
 
           <Spacer mb={2} />
 
-          {enableCreateAlert && savedSearchEntity && (
+          {savedSearchEntity && (
```
Comment (gkartalis) [inline src/v2/Components/ArtworkFilter/index.tsx:278]: **suggestion:** what do you think of changing that to 
```suggestion
          {!!savedSearchEntity && (
```
just to make sure that nothing weird will happen. not saying that this won't work in react but I think it is better to explicitly make it to a boolean over a falsy value, what do you think?

File: src/v2/Components/ArtworkFilter/index.tsx [hunk 2]
```diff
@@ -335,10 +336,11 @@ export const BaseArtworkFilter: React.FC<
             // Safe to remove once artwork masonry uses CSS grid.
             width="100%"
           >
-            {enableCreateAlert && savedSearchEntity && (
+            {savedSearchEntity && (
```
Comment (gkartalis) [inline src/v2/Components/ArtworkFilter/index.tsx:339]: same here as above

File: src/v2/Components/SavedSearchAlert/Components/__tests__/SavedSearchAlertArtworkGridFilterPills.jest.tsx [hunk 3]
```diff
@@ -78,4 +80,70 @@ describe("SavedSearchAlertArtworkGridFilterPills", () => {
 
     expect(screen.getByText("Banksy")).toBeInTheDocument()
   })
+
+  it("display followed artist pills when 'Artists I Follow' is selected", () => {
```
Comment (gkartalis) [inline src/v2/Components/SavedSearchAlert/Components/__tests__/SavedSearchAlertArtworkGridFilterPills.jest.tsx:84]: **praise:** thanks for the consistent great test cases as always!

**Review by gkartalis — APPROVED** [review]

File: src/v2/Apps/Partner/Routes/__tests__/Works.jest.tsx [hunk 4]
```diff
@@ -122,4 +142,63 @@ describe("PartnerArtworks", () => {
       expect(filterWrappers.at(filterIndex).prop("expanded")).toEqual(expanded)
     })
   })
+
+  it("should render selected filters as pills", () => {
```
Comment (gkartalis) [inline src/v2/Apps/Partner/Routes/__tests__/Works.jest.tsx:146]: **praise:** 👏

File: src/v2/Components/SavedSearchAlert/Components/__tests__/SavedSearchAlertArtworkGridFilterPills.jest.tsx (hunk 3 above)
Comment (dimatretyak) [reply src/v2/Components/SavedSearchAlert/Components/__tests__/SavedSearchAlertArtworkGridFilterPills.jest.tsx:84]: Thanks!

File: src/v2/Apps/Settings/Routes/SavedSearchAlerts/Components/SavedSearchAlertEditForm.tsx [hunk 5]
```diff
@@ -385,14 +385,7 @@ const SAVED_SEARCH_ALERT_EDIT_FORM_QUERY = graphql`
...
       artistIDs: $artistIds
-      aggregations: [
-        ARTIST
-        LOCATION_CITY
-        MATERIALS_TERMS
-        MEDIUM
-        PARTNER
-        COLOR
-      ]
+      aggregations: [LOCATION_CITY, MATERIALS_TERMS, MEDIUM, PARTNER, COLOR]
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/v2/Apps/Settings/Routes/SavedSearchAlerts/Components/SavedSearchAlertEditForm.tsx:388]: We don't need to use data from the `ARTIST` aggregation here, since we get information about artists [here](https://github.com/artsy/force/blob/3742a208af183454173f22e7d19511d80773ce0d/src/v2/Apps/Settings/Routes/SavedSearchAlerts/Components/SavedSearchAlertEditForm.tsx#L344-L354)

**Review by gkartalis — DISMISSED** [review]

File: src/v2/Components/ArtworkFilter/index.tsx [hunk 6]
```diff
@@ -275,10 +275,11 @@ export const BaseArtworkFilter: React.FC<
 
           <Spacer mb={2} />
 
-          {enableCreateAlert && savedSearchEntity && (
+          {!!savedSearchEntity && (
```
Comment (mdole) [inline src/v2/Components/ArtworkFilter/index.tsx:278]: @dimatretyak I think we could do a little more "separation of concerns" here. Instead of rendering both filter pills and the "create alert" button when a `savedSearchEntity` is passed, what do you think about separating them so that rendering filter pills is based on a `renderFilterPills` prop and the button on a `renderCreateAlertButton` prop?

File: src/v2/Apps/Partner/Routes/Works/index.tsx [hunk 7]
```diff
@@ -24,6 +26,17 @@ export const Artworks: React.FC<PartnerArtworkFilterProps> = ({
...
+    placeholder: "Partner Page", // TODO: Clarify and update it later
+    defaultArtists: [],
+    owner: {
+      type: OwnerType.partner,
+      slug: partner.slug,
+      id: partner.internalID,
+      name: partner.name ?? "",
+    },
+  }
+
```
Comment (mdole) [inline src/v2/Apps/Partner/Routes/Works/index.tsx:39]: **minor, non-blocking:** suggest removing this and moving it to a separate PR since it's not necessarily related to rendering the filter pills (see my other comment about making those separate concerns)

**Review by mdole — COMMENTED** [review]
really nice work @dimatretyak! Left a suggestion about splitting out the logic for rendering pills vs. rendering the create alert button, let me know what you think 🙂

File: src/v2/Components/ArtworkFilter/index.tsx (hunk 6 above)
Comment (dimatretyak) [reply src/v2/Components/ArtworkFilter/index.tsx:278]: In theory, this is a good idea 🤔. But it seems to me that such changes will only complicate our logic and we will often have to write the duplicate code.

Comment (dimatretyak) [reply src/v2/Components/ArtworkFilter/index.tsx:278]: If it doesn't bother you, can you give an example of the benefits that we will get if we add these changes?

Comment (dimatretyak) [reply src/v2/Components/ArtworkFilter/index.tsx:278]: PR #10391 (simpler solution)
PR #10391 (a more complex solution with duplication of information in some places, but it gives us more flexibility)

File: src/v2/Apps/Partner/Routes/Works/index.tsx [hunk 8]
```diff
@@ -24,6 +26,17 @@ export const Artworks: React.FC<PartnerArtworkFilterProps> = ({
 }) => {
   const { sidebar } = partner
 
+  const savedSearchEntity: SavedSearchEntity = {
+    placeholder: "Partner Page", // TODO: Clarify and update it later
```
Comment (nickskalkin) [inline src/v2/Apps/Partner/Routes/Works/index.tsx:30]: Don't forget to address this comment :)

**Review by nickskalkin — APPROVED** [review]
looks good to me 👍

**Review by nickskalkin — APPROVED** [review]
